src/hooks/charts/useChartDataLabels.ts

```diff
@@ -9,9 +9,16 @@ export default function useChartDataLabels({ asset, chartType, points }: any) {
 
   const getPercentChangeForPrice = useCallback(
     (startPrice: number) => {
-      const endPrice = points?.[points.length - 1].y;
-      const percent = ((endPrice - startPrice) / startPrice) * 100;
-      return formatPercentChange(percent);
+      try {
+        const endPrice = points?.[points.length - 1].y;
+        const percent = ((endPrice - startPrice) / startPrice) * 100;
+        return formatPercentChange(percent);
+      } catch {
+        // couldn't repro but sentry caught `y undefined in points?.[points.length - 1]`
+        // apparently it happens when zerion is starting to collect price data for this new token
+        // Chart checks if latestPrice is NaN later that's why Im returing NaN
+        return NaN;
+      }
```

Comment:
the code originally use `?` optional chaining to avoid throw on line 13, let's keep it that way by adding one more question mark.

However there is another problem in  the code, it was not typed correctly, when `percent` is an `NaN`, `formatPercentChange(percent)` will return string `'NaN'` instead of a `NaN`.

With the original modification in this PR the actual output of `latestChange` can be 3 types: 
1. number, the changes
2. string `'NaN'`, when `points` is `undefined or null`, the NaN gets formatted into a string
3. the actual `NaN`

Recommand below changes to simply it:

```suggestion
        const endPrice = points?.[points.length - 1]?.y;
        const percent = ((endPrice - startPrice) / startPrice) * 100;
        
        if (isNaN(percent)) {
          return NaN
        }
        
        return formatPercentChange(percent);
```

also recommand adding output type to make it clear what it outputs.

Reply:
oh you're right optional chaining is a way cleaner solution, good call